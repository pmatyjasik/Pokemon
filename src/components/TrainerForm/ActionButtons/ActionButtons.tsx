import { Box, Button } from "@mui/material";

interface ActionButtonsProps {
    onReset: () => void;
}

export const ActionButtons = ({ onReset }: ActionButtonsProps) => (
    <Box display='flex' justifyContent='flex-end' gap={4} width='100%'>
        <Button variant='contained' color='info' onClick={onReset}>
            Reset
        </Button>
        <Button variant='contained' color='primary' type='submit'>
            Submit
        </Button>
    </Box>
);
