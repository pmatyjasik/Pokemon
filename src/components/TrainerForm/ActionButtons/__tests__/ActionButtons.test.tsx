import React from "react";
import { render, screen } from "@testing-library/react";
import { ActionButtons } from "@/components/TrainerForm/ActionButtons/ActionButtons";

const renderComponent = () => {
    return render(<ActionButtons onReset={() => {}} />);
};

describe("ActionButtons", () => {
    it("Matches DOM Snapshot", () => {
        const { asFragment } = renderComponent();

        expect(asFragment()).toMatchSnapshot();
    });

    test("renders Reset and Submit buttons", () => {
        renderComponent();

        expect(screen.getByText("Reset")).toBeInTheDocument();
        expect(screen.getByText("Submit")).toBeInTheDocument();
    });

    test("submit button has type submit", () => {
        renderComponent();

        expect(screen.getByText("Submit")).toHaveAttribute("type", "submit");
    });
});
