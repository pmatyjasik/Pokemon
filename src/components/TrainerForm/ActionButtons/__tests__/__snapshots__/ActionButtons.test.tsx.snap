// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionButtons Matches DOM Snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-1jf6bot"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedInfo MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorInfo MuiButton-root MuiButton-contained MuiButton-containedInfo MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorInfo css-kjirbw-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Reset
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-sghohy-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="submit"
    >
      Submit
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
